libs/contrib/Data/String/Parser.idr

```diff
@@ -183,3 +185,33 @@ token s = lexeme (skip (string s)) <?> "token " ++ show s
 export
 fail : Monad m => String -> ParseT m a
 fail x = P $ \s => do pure $ Fail s.pos x
+
+export
+digit : Parser (Fin 10)
+digit = do x <- satisfy isDigit
+           case lookup x digits of
+                Nothing => P $ \s => do pure $ Fail s.pos "not a digit"
+                Just y => P $ \s => Id (OK y s)
+  where
+    digits : List (Char, Fin 10)
+    digits = [ ('0', FZ)
```

Comment:
We have overloading of literals for `Fin` (cf. `Data.Fin.fromInteger`) so
you should even be able to write `[ ('0', 0), ('1', 1), ... ]`

Reply:
Sweet! Updated.